vmngclient/api/mtt_aaa_api.py

```diff
@@ -0,0 +1,186 @@
+import logging
+
+from vmngclient.dataclasses import TenantAAA, TenantRadiusServer, TenantTacacsServer
+from vmngclient.session import vManageSession
+from vmngclient.utils.creation_tools import asdict, create_dataclass
+
+logger = logging.getLogger(__name__)
+
+
+class AAAConfigNotPresent(Exception):
+    pass
+
+
+class TenantAaaAPI:
+    """
+    Used to configure  mtt tenant management users remote servers
+    """
+
+    def __init__(self, session: vManageSession) -> None:
+        self.session = session
+
+    def __str__(self) -> str:
+        return str(self.session)
+
+    def aaa_exists(self) -> bool:
+        url_path = "/dataservice/admin/aaa"
+        if not self.session.get_data(url_path):
+            return False
+        else:
+            return True
+
+    def add_aaa(self, tenant_aaa: TenantAAA) -> bool:
+        """ "
+        TenantAAA:
+            "authOrder": [ "local", "radius", "tacacs"],
+            "authFallback": true,
+            "adminAuthOrder": false,
+            "auditDisable": false,
+            "accounting": false,
+            "radiusServers": "server1"
+
+        returns bool depending on the api post call
+        """
+        url_path = "/dataservice/admin/aaa"
+        data = asdict(tenant_aaa)  # type: ignore
+        response = self.session.post(url=url_path, json=data)
+        return True if response.status_code == 200 else False
+
+    def get_aaa(self) -> TenantAAA:
+        """
+        Returns the Tenant AAA
+        :param aaa:
+        :return:
+        """
+        url_path = "/dataservice/admin/aaa"
+        tenant_aaa = self.session.get_data(url_path)
+        logger.debug(f"Tenant AAA: {tenant_aaa}")
+        return create_dataclass(TenantAAA, tenant_aaa)
+
+    def del_aaa(self) -> bool:
+        """
+        Delete aaa works only for tenants
+        :return:
+        """
+        url_path = "/dataservice/admin/aaa"
+        logger.debug(f"Deleting AAA on {self.session.get_tenant_id()}.")
+        response = self.session.delete(url_path)
+        if not self.aaa_exists():
+            raise AAAConfigNotPresent("No Aaa config present for Tenant")
+        return True if response.status_code == 200 else False
+
+    def put_aaa(self, tenant_AAA: TenantAAA) -> bool:
+        """
+        Updated the AAA for tenant
+        :return:
+        """
+        url_path = "/dataservice/admin/aaa"
+        data = asdict(tenant_AAA)  # type: ignore
+        response = self.session.put(url=url_path, json=data)
+        return True if response.status_code == 200 else False
+
+
+class TenantRadiusAPI:
+    """
+    Used to configure  mtt tenant remote aaa radius servers
+    """
+
+    def __init__(self, session: vManageSession) -> None:
+        self.session = session
+
+    def __str__(self) -> str:
+        return str(self.session)
+
+    def add_radius(self, radius_server: TenantRadiusServer) -> bool:
```

Comment:
```suggestion
    def add(self, radius_server: TenantRadiusServer) -> bool:
```

Reply:
public API should be each time simplified if possible

Reply:
But this api file is addressing three components AAA, RADIUS and TACACS